src/lib/discord/bot.ts

```diff
@@ -13,13 +17,261 @@ new Command({
+			return {
+				content: `Requested entry "${name}" does not exist, or is unavailable.`,
+				color: 0x650ac9,
+			}
+		}
+
+		let content: string
+
+		if (entry.content.length > 985) {
+			content = `${entry.content.slice(0, 985)}...`
+		} else {
+			content = entry.content
+		}
 		return {
 			embeds: [
 				{
-					title: `Entry "${i.options.name}"`,
+					title: name,
+					url: `${BASE_URL}/entry/${entry.slug}`,
+					description: content,
+					color: 0x650ac9,
 				},
 			],
 		}
 	})
 
+new Command({
+	name: 'random',
+	description: 'View a random Repldex entry',
+}).handle(async data => {
+	const entryCount: number = await countEntries()
+	const entries = await fetchEntries({
+		limit: 1,
+		skip: Math.floor(Math.random() * entryCount),
+	})
+	const entry = entries[Math.floor(Math.random() * entries.length)]
+
+	return {
+		embeds: [
+			{
+				title: entry.title,
+				url: `${process.env.BASE_URL}/entry/${entry.slug}`,
+				description: entry.content,
+				color: 0x650ac9,
+			},
+		],
+	}
+})
+
+// This command has like 2 million issues, so i have temporaily removed it!
+
+new Command({
+	name: 'search',
+	description: 'Search Repldex entries',
+})
+	.addOption({
+		name: 'query',
+		description: 'Query to search for',
+		type: ApplicationCommandOptionType.String,
+		required: true,
+	})
+	.handle(async data => {
+		console.log(data)
+		//query should always be a string. probably a better way to do this
+		if (typeof data.options.query !== 'string') throw new Error('Custom ID is invalid')
+		const query: string = data.options.query
+		const entries = await fetchEntries({
+			query: query,
+			limit: 10,
+			skip: 0,
+		})
+		const embed: APIEmbed = {
+			title: `Search Results for ${query}`,
+			fields: [],
+			footer: {
+				text: 'Page 1',
+			},
+		}
+		for (const entry of entries) {
+			embed.fields!.push({
+				name: entry.title,
+				value: `[Link](${process.env.BASE_URL}/entry/${entry.slug})`,
+			})
+		}
+		//note: as of now buttons just sit around. clicking on them does not do anything
+		return {
+			components: [
+				{
+					//buttons must be in an action row.... why? blame discord
+					type: 1,
+					components: [
+						{
+							//type 2 = button
+							type: 2,
+							style: 1,
+							label: 'Back',
+							emoji: {
+								id: null,
+								name: '◀️',
+							},
+							//command-button-page-query
+							custom_id: `search-back-1-${query}`,
+						},
+						{
+							type: 2,
+							style: 1,
+							label: 'Forward',
+							emoji: {
+								id: null,
+								name: '▶️',
+							},
+							custom_id: `search-forward-1-${query}`,
+						},
+					],
+				},
+			],
+			embeds: [embed],
+		}
+	})
+	.handleComponents(async args => {
+		const action = args[0]
+		//`+` is unary operator that converts to int
+		let page: number = +args[1]
+		const query = args[2]
+		let entries = await fetchEntries({
+			query: query,
+			limit: 0,
+			skip: (page - 1) * 10,
+		})
+		entries = entries.slice(0, 10)
```

Comment:
Setting the limit to 0 fetches every entry, which is bad. It should just be 10. Also that makes the `entries = entries.slice(0, 10)` unnecessary since it's already limiting it for you.